homeassistant/components/analytics/analytics.py

```diff
@@ -306,3 +337,37 @@ async def send_analytics(self, _: datetime | None = None) -> None:
             LOGGER.error(
                 "Error sending analytics to %s: %r", ANALYTICS_ENDPOINT_URL, err
             )
+
+    @callback
+    def _async_validate_integration(
```

Comment:
This was split out due to pre_commit complaining about complexity, which I kinda agree with.

Reply:
Let's rename this to better explain what is being validated
```suggestion
    def _should_report_integration(
```